tagless-final/src/main/java/com/github/jorgecastillo/kotlinandroid/tagless/algebras/business/HeroesUseCases.kt

```diff
@@ -1,21 +1,17 @@
 package com.github.jorgecastillo.kotlinandroid.tagless.algebras.business
 
-import arrow.HK
-import arrow.TC
-import arrow.typeclass
+import arrow.Kind
 import com.github.jorgecastillo.kotlinandroid.tagless.algebras.persistence.HeroesRepository
 import com.karumi.marvelapiclient.model.CharacterDto
+import javax.inject.Inject
 
-@typeclass
-interface HeroesUseCases<F> : TC {
+class HeroesUseCases<F> @Inject constructor(val repository: HeroesRepository<F>) {
```

Comment:
I guess this is how a typeclass contract that can be injected for any parametric data type `F` would look like now, and can easily be injected by Dagger. But at some point I feel like there is a disconnection with the typeclass concept, since devs don't find any tags or annotations letting them know about what is it 🤔. Just as a thought.

Meanwhile, and for didactic purposes which is the goal of this repo, could you add Kotlindoc on top of the class clarifying what is this? Also over all the typeclasses you are defining on this PR.

Or maybe using naming conventions for readability: `UseCasesTypeClass`

Reply:
> Meanwhile, and for didactic purposes which is the goal of this repo, could you add Kotlindoc on top of the class clarifying what is this? Also over all the typeclasses you are defining on this PR.

👍 I think in an example repo like this some Kdoc would be really useful to make it easier to navigate for people learning these concepts.

I am not sure I follow your other comment. What do you mean with the following?
>  But at some point I feel like there is a disconnection with the typeclass concept, since devs don't find any tags or annotations letting them know about what is it